src/utils/thread_helper.cpp

```diff
@@ -1694,5 +1668,43 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
     return error;
 }
 #endif // OTBR_ENABLE_TELEMETRY_DATA_API
+
+otError ThreadHelper::ProcessDatasetForMigration(otOperationalDatasetTlvs &aDatasetTlvs, uint32_t aDelayMilli)
+{
+    otError  error = OT_ERROR_NONE;
+    Tlv     *tlv;
+    timespec currentTime;
+    uint64_t pendingTimestamp = 0;
+
+    VerifyOrExit(FindTlv(OT_MESHCOP_TLV_PENDINGTIMESTAMP, aDatasetTlvs.mTlvs, aDatasetTlvs.mLength) == nullptr,
+                 error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(FindTlv(OT_MESHCOP_TLV_DELAYTIMER, aDatasetTlvs.mTlvs, aDatasetTlvs.mLength) == nullptr,
+                 error = OT_ERROR_INVALID_ARGS);
+
+    // There must be sufficient space for a Pending Timestamp TLV and a Delay Timer TLV.
+    VerifyOrExit(
+        static_cast<int>(aDatasetTlvs.mLength +
+                         (sizeof(uint8_t) + sizeof(uint8_t) + sizeof(uint64_t))    // Pending Timestamp TLV (10 bytes)
+                         + (sizeof(uint8_t) + sizeof(uint8_t) + sizeof(uint32_t))) // Delay Timer TLV (6 bytes)
+            <= int{sizeof(aDatasetTlvs.mTlvs)},
+        error = OT_ERROR_INVALID_ARGS);
+
+    tlv = reinterpret_cast<Tlv *>(aDatasetTlvs.mTlvs + aDatasetTlvs.mLength);
+    tlv->SetType(OT_MESHCOP_TLV_PENDINGTIMESTAMP);
+    clock_gettime(CLOCK_REALTIME, &currentTime);
+    pendingTimestamp |= (static_cast<uint64_t>(currentTime.tv_sec) << 16);
+    pendingTimestamp |= (((static_cast<uint64_t>(currentTime.tv_nsec) * 32768 / 1000000000) & 0x7fff) << 1);
+    tlv->SetValue(pendingTimestamp);
+
+    tlv = tlv->GetNext();
+    tlv->SetType(OT_MESHCOP_TLV_DELAYTIMER);
+    tlv->SetValue(aDelayMilli);
```

Comment:
Is the endian-ness correctly handled when setting a `uint32_t` value on TLV?

Reply:
I've checked and it's handled (written in big endian). These [code](https://github.com/openthread/ot-br-posix/blob/main/src/common/tlv.hpp#L155) were in introduced since this PR: https://github.com/openthread/ot-br-posix/pull/1051

I think the code doesn't reflect that the endianness is handled correctly because in ot-br-posix the tlv handling code isn't that complete as that in OT. Let's further improve this in future PRs.

Reply:
Thanks. SGTM.